### PR TITLE
Add environment support to the discover step

### PR DESCRIPTION
**Pull request by psss**
Combine L1 and L2 environment in Test.export().
Include a simple test with an environment variable.
Extend utils.format() to support dictionaries as well.

### CODE REVIEW

**Comment by psss** [comment]
Any test with environment variable set still fails because of problem mentioned in #99.

**Comment by pvalena** [comment]
LGTM.

**Comment by psss** [comment]
@pvalena, thanks for the review. @thrix, I had to disable the new test because of missing environment support in cruncher. Seems we will be running into similar issues more and more often. Would be nice to replace cruncher with tmt (relatively) soon.

File: tmt/base.py [hunk 1]
```diff
@@ -210,7 +210,13 @@ def export(self, format_='yaml', keys=None):
...
         if self.duration is not None:
             data['duration'] = self.duration
-        if self.environment is not None:
+        # Combine environment variables (plan overrides test)
+        if self.environment is not None or environment is not None:
+            combined_environment = dict()
+            if self.environment:
+                combined_environment.update(self.environment)
+            if environment:
+                combined_environment.update(environment)
```
Comment (thrix) [inline tmt/base.py]: @psss  What about debug output of the combined environment, so user has change to understand what happened here for troubleshooting?

Comment (thrix) [reply tmt/base.py]: ok, we can do that later :D

**Comment by pvalena** [comment]
> Seems we will be running into similar issues more and more often. Would be nice to replace cruncher with tmt (relatively) soon.

@psss for my information, what's blocking that (from tmt side)?

**Comment by psss** [comment]
> @psss for my information, what's blocking that (from tmt side)?

Would be good to improve the output/logging/results presentation. But I think the main point is just the priority. The cruncher is somewhat stable and changing that would require extra time and effort. We just need to prioritize this well.

**Comment by psss** [comment]
All tests passed, just the results are not updated here. Let's give it one more attempt.

**Comment by psss** [comment]
All tests passed. There is a problem with Packit results reporting. Merging.

**Pull request merged**